specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json

```diff
@@ -0,0 +1,2618 @@
+                    "Monday",
+                    "Tuesday",
+                    "Wednesday",
+                    "Thursday",
+                    "Friday",
+                    "Saturday",
+                    "Sunday"
+                  ],
+                  "x-ms-enum": {
+                    "name": "MaintenanceDayOfWeekType",
+                    "modelAsString": false
+                  },
+                  "description" : "Day of the week that the maintenance window will occur"
+                },
+                "timeOfDay" : {
+                  "type" : "string",
+                  "pattern": "^([0-9]|0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$",
+                  "description": "The time of day to start the maintenance window"
+                }
+              }
+            },
+            "hsmSettings": {
+              "$ref": "#/definitions/amlFilesystemHSMSettings",
+              "description": "Specifies HSM settings of the amlFilesystem."
+            },
+            "namespaceHydrationStatus": {
+              "description" : "Namespace Hydration Status during creation",
+              "$ref": "#/definitions/amlFilesystemNamespaceHydrationStatus"
+            },
+            "archiveSettings" : {
+              "description": "Archive settings and status",
+              "$ref" :  "#/definitions/amlFilesystemArchive"
+            }
+          },
+          "type": "object"
+        },
+        "sku": {
+          "description": "SKU for the amlFilesystem",
+          "properties": {
+            "name": {
+              "description": "SKU name for this amlFilesystem",
+              "type": "string"
+            }
+          },
+          "type": "object"
+        }
+      },
+      "x-ms-azure-resource": true,
+      "type": "object"
+    },
+    "amlFilesystemName" : {
+      "description": "Name for the amlFilesystem.  Alphanumerics, underscores, and hyphens. Start and end with alphanumeric.",
+      "type" : "string",
+      "pattern": "^[0-9a-zA-Z][-0-9a-zA-Z_]{0,78}[0-9a-zA-Z]$",
+      "minLength": 2,
+      "maxLength": 80
+    },
+    "amlFilesystemsListResult": {
+      "description": "Result of the request to list Caches. It contains a list of Caches and a URL link to get the next set of results.",
+      "properties": {
+        "nextLink": {
+          "description": "URL to get the next set of Cache list results, if there are any.",
+          "type": "string"
+        },
+        "value": {
+          "description": "List of Caches.",
+          "items": {
+            "$ref": "#/definitions/amlFilesystem"
+          },
+          "type": "array"
+        }
+      },
+      "type": "object"
+    },
+    "amlFilesystemHealth": {
+      "description": "An indication of amlFilesystem health. Gives more information about health than just that related to provisioning.",
+      "properties": {
+        "state": {
+          "description": "List of amlFilesystem health states.",
+          "enum": [
```

Comment:
To support polling at startup, we'll need an additional state to indicate that the filesystem is provisioned but still hydrating the namespace- perhaps something like HydratingNamespace.

Reply:
Added.